### PR TITLE
Add MX floating point types (f4E2M1FN, f6E2M3FN, f6E3M2FN, f8E8M0FNU)

### PR DESCRIPTION
**Pull request by sergey-kozub**
This PR is adding MX (microscaling) floating point types to StableHLO.

The RFC is created in https://github.com/openxla/stablehlo/pull/2581

Bumps VHLO version to 1.8.0

### CODE REVIEW

**Review by sdasgup3 — APPROVED** [review]

**Review by GleasonK — CHANGES_REQUESTED** [review]
adding blocking review until RFC merged, will re-review once RFC is in

File: stablehlo/dialect/VhloBytecode.cpp [hunk 1]
```diff
@@ -189,7 +189,7 @@ enum AttributeCode {
 /// location is updated.
 enum TypeCode {
   // TO ADD TYPE: Add an enum value with doc string for new type.
-  // Next available code: 37
+  // Next available code: 40
```
Comment (GleasonK) [inline stablehlo/dialect/VhloBytecode.cpp:192]: 41

` kFloatF8E8M0FNUV1Type = 40,`

File: build_tools/llvm_version.txt [hunk 2]
```diff
@@ -1 +1 @@
-00128a20eec27246719d73ba427bf821883b00b4
+3f9cabae0029bcbe88835aaa4c417ce41e584fb1
```
Comment (GleasonK) [inline build_tools/llvm_version.txt:1]: Revert? I'd guess we're ahead of this pin by now.

File: WORKSPACE.bazel [hunk 3]
```diff
@@ -17,9 +17,9 @@ workspace(name = "stablehlo")
 
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 
-LLVM_COMMIT = "00128a20eec27246719d73ba427bf821883b00b4"
+LLVM_COMMIT = "3f9cabae0029bcbe88835aaa4c417ce41e584fb1"
 
-LLVM_SHA256 = "9fff2ccb6c262f3d5e2f98c281a0b99a585daee83742e1599709ff61cfc222af"
+LLVM_SHA256 = "626e1c8e491cd70ef540c403fdd43b9ff9ee50aafe181c32b14e67f715ad015e"
```
Comment (GleasonK) [inline WORKSPACE.bazel:22]: Revert? I'd guess we're ahead of this pin by now.

**Review by GleasonK — APPROVED** [review]

File: stablehlo/dialect/VhloBytecode.cpp (hunk 1 above)
Comment (sergey-kozub) [reply stablehlo/dialect/VhloBytecode.cpp]: Done, thanks.

File: build_tools/llvm_version.txt (hunk 2 above)
Comment (sergey-kozub) [reply build_tools/llvm_version.txt]: Done.

File: WORKSPACE.bazel (hunk 3 above)
Comment (sergey-kozub) [reply WORKSPACE.bazel]: Done.

**Review by GleasonK — APPROVED** [review]

**Pull request merged**